staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -57,7 +57,8 @@ func GetTransformerOverrides(filepath string) (map[schema.GroupResource]value.Tr
 
 // ParseEncryptionConfiguration parses configuration data and returns the transformer overrides
 func ParseEncryptionConfiguration(f io.Reader) (map[schema.GroupResource]value.Transformer, error) {
-	configFileContents, err := ioutil.ReadAll(f)
+	// Upper limit of 100 KB on file size.
+	configFileContents, err := ioutil.ReadAll(io.LimitReader(f, 102400))
```

Comment:
The problem with that function is that it "stops with EOF after n bytes", so there it's impossible to differentiate situation when the file is bigger than 100Kb of when it's exactly equals to 100Kb. If a file has 101Kb and you read only 100Kb later you will fail with a strange error during deserialization to YAML or even worst -- you will have only first bytes of the encryption key and you couldn't decode values from the database (and it will be really hard to find out why).